pom.xml

```diff
@@ -168,6 +187,11 @@
 			<artifactId>junit-vintage-engine</artifactId>
```

Comment:
Wouldn't you need to remove this dependencies to not rely on JUnit 4?

Reply:
I think we still have a lot of tests running on JUnit4 so it wouldn't make sense to remove this. To ensure that the tests have been successfully migrated to JUnit5, for every test after migration, I remove this dependency from the code and run the test. 
Does this make sense or is there a better way to go about it?

Reply:
Sure, as  long as there is a way to tell that the migrated tests are running with JUnit 5, and not 4.

Reply:
That's a good validation! To make extra sure there is no JUnit4 left in the project, you can check with `mvn dependency:tree -Dincludes=junit:junit`. I think you'd have to remove both `junit:junit` and `org.junit.vintage:junit-vintage-engine`, since the latter has a transitive dependency on the former.

Reply:
Okay, to answer the original question -- `org.junit.vintage:junit-vintage-engine` uses `junit:junit` for actual implementation. From [docs](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven-engines-configure): "Maven Surefire and Maven Failsafe can run JUnit 4 based tests alongside Jupiter tests as long as you configure test scoped dependencies on JUnit 4 and the JUnit Vintage TestEngine".

So we have to keep `junit:junit` dependency until we have migrated all Junit4 tests to Junit5.